### PR TITLE
Revamp

### PR DESCRIPTION
**Pull request by Moonianne**
###### Change Set:
 - Removed tracked files in `.idea` that weren't intended
 - Rework how user interactions result in `UserIntent`s (experimenting: subject to change based on ease of testability)
 - Rework View layout  because the old one was ugly!
 - No long use a "calculate" button, all interactions with view result in a calculation once one is possible
 - Rename `MainFragment` to `CalculateFragment`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,4 @@
 .externalNativeBuild
 .cxx
 local.properties
-/.idea
+**/.idea**
```
Comment (kelveenfabian) [inline .gitignore:10]: Why the two stars?

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/intent/MainIntent.kt [hunk 2]
```diff
@@ -8,6 +8,4 @@ sealed class MainIntent {
         val hasDryFood: Boolean,
         val hasGreenie: Boolean,
     ) : MainIntent()
```
Comment (kelveenfabian) [inline app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/intent/MainIntent.kt:10]: MainIntent.kt is inheriting MainIntent()?

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/view/CalculateFragment.kt [hunk 3]
```diff
@@ -1,39 +1,45 @@
...
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
 
 class CalculateFragment : Fragment() {
-
-    private lateinit var viewModel: MainViewModel
-    private lateinit var binding: MainFragmentBinding
+    private lateinit var viewModel: CalculateViewModel
+    private lateinit var binding: CalculateFragmentBinding
```
Comment (kelveenfabian) [inline app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/view/CalculateFragment.kt:27]: Isn't this supposed to take you to an XML when you cmd + click on it?

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewmodel/CalculateViewModel.kt [hunk 4]
```diff
@@ -13,7 +13,7 @@ import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.consumeAsFlow
 import kotlinx.coroutines.launch
 
-class MainViewModel(private val calculator: DoggyMealCalculator) : ViewModel() {
+class CalculateViewModel(private val calculator: DoggyMealCalculator) : ViewModel() {
     val userIntent = Channel<MainIntent>(Channel.UNLIMITED)
     private val viewState = MutableStateFlow<MainState>(MainState.Init())
     val state: StateFlow<MainState> get() = viewState
```
Comment (kelveenfabian) [inline app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewmodel/CalculateViewModel.kt:19]: State is manipulated through the view models?!

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewUtil/ViewExtensions.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package com.gerardojim.nuggetscalculator.ui.main.viewUtil
```
Comment (kelveenfabian) [inline app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewUtil/ViewExtensions.kt:1]: Okay this makes sense.

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/intent/MainIntent.kt (hunk 2 above)
Comment (Moonianne) [reply app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/intent/MainIntent.kt:10]: @kelveenfabian 
```
sealed class MainIntent {
    data class Calculate(
        val selectedFoodPosition: Int,
        val dailyCaloricTarget: Int,
        val offset: Double = 0.0,
        val hasDryFood: Boolean,
        val hasGreenie: Boolean,
    ) : MainIntent()
}

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/view/CalculateFragment.kt (hunk 3 above)
Comment (Moonianne) [reply app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/view/CalculateFragment.kt:27]: @kelveenfabian is it? You might be thinking of databinding

File: .gitignore (hunk 1 above)
Comment (Moonianne) [reply .gitignore:10]: lol still trying to get this to work. I want it to never track anything in that package or anything that includes that name

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewmodel/CalculateViewModel.kt (hunk 4 above)
Comment (Moonianne) [reply app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewmodel/CalculateViewModel.kt:19]: Techincally. The ViewModel spits out a ViewState

File: app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewUtil/ViewExtensions.kt (hunk 5 above)
Comment (Moonianne) [reply app/src/main/java/com/gerardojim/nuggetscalculator/ui/main/viewUtil/ViewExtensions.kt:1]: Like it?

**Pull request merged**